Parser/pegen/pegen.c

```diff
@@ -878,8 +897,7 @@ _PyPegen_Parser_New(struct tok_state *tok, int start_rule, int *errcode, PyArena
 {
     Parser *p = PyMem_Malloc(sizeof(Parser));
     if (p == NULL) {
-        PyErr_Format(PyExc_MemoryError, "Out of memory for Parser");
-        return NULL;
+        return (void *) PyErr_NoMemory();
```

Comment:
If you cast explicitly, I would prefer to cast to `Parser*`. Same coment for other error paths below.